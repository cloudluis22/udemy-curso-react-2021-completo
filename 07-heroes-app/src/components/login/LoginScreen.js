import React, { useContext } from 'react';
import AuthContext from '../../auth/AuthContext';
import types from '../../types/types';

export const LoginScreen = ({ history }) => {

    const { dispatch } = useContext(AuthContext)

    const handleLogin = () => {

       const lastPath = localStorage.getItem('lastPath') || '/';

       const newUser = {
            name: 'Sesi√≥n Iniciada',
            logged: true
        }

        const action = {
            type: types.login,
            payload: newUser
        }

        dispatch(action);

        history.replace(lastPath);

    };

    return (
        <div className="container mt-5">

            <h1>Login Screen</h1>
            <hr />

            <button
                className="btn btn-primary"
                onClick={ handleLogin }
            >
                Login
            </button>

        </div>
    )
}
