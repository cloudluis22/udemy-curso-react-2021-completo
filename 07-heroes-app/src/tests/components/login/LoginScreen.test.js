import React from 'react';
import { mount } from 'enzyme';
import { LoginScreen } from '../../../components/login/LoginScreen';
import AuthContext from '../../../auth/AuthContext';

describe('Pruebas en el componente de "<LoginScreen />".', () => {

    const contextValue = {
        user: 'José Luis',
        dispatch: jest.fn()
    }

    const historyMock = {
        replace: jest.fn()
    }

    const wrapper = mount(

        <AuthContext.Provider value={ contextValue }>
            <LoginScreen history={ historyMock } />
        </AuthContext.Provider>

    );

    test('Debe de mostrarse correctamente y generar la snapshot.', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de realizar el dispatch del reducer y la navegación.', () => {

        wrapper.find('button').prop('onClick')();

        expect(contextValue.dispatch).toHaveBeenCalled();
        expect(historyMock.replace).toHaveBeenCalled();

    });

});