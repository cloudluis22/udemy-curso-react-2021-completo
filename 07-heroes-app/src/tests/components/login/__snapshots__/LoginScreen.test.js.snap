// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente de "<LoginScreen />". Debe de mostrarse correctamente y generar la snapshot. 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
